connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/PostgresIntegrationSuite.scala

```diff
@@ -128,7 +128,7 @@ class PostgresIntegrationSuite extends DockerJDBCIntegrationSuite {
       "('abcd', 'efgh', 'ijkl', 'mnop', 'q')").executeUpdate()
 
     conn.prepareStatement("CREATE TABLE char_array_types (" +
-      "c0 char(4)[], c1 character(4)[], c2 character varying(4)[], c3 varchar(4)[], c4 bpchar[])"
+      "c0 char(4)[], c1 character(4)[], c2 character varying(4)[], c3 varchar(4)[], c4 bpchar(1)[])"
```

Comment:
According to the pg's doc, char w/o length specifier is char(1). While based on my offline testing with pgjdbc, the precision here is int.max, which causes the tests fail with OOM.

Reply:
```scala
val rs = stat.executeQuery("select * from char_array_types")
rs.getMetaData.getPrecision(5)
res0: Int = 2147483647
```

Reply:
updated